src/main/scala/com/fulcrumgenomics/bam/Bams.scala

```diff
@@ -279,4 +279,28 @@ object Bams extends LazyLogging {
 
     (min(firstEnd,secondEnd), max(firstEnd,secondEnd))
   }
+
+
+  /** If the read is mapped in an FR pair, returns the distance of the position from the other end
+    * of the template, other wise returns None.
+    *
+    * @param rec the SAMRecord whose insert to calculate the position within
+    * @param genomicPosition the genomic position of interest (NOT the position within the read)
+    */
+  def positionFromOtherEndOfTemplate(rec: SAMRecord, genomicPosition: Int): Option[Int] = {
+    if (isFrPair(rec) && rec.getInferredInsertSize != 0) {
+      val isize        = rec.getInferredInsertSize
+      val thisEnd      = if (rec.getReadNegativeStrandFlag) rec.getAlignmentEnd else rec.getAlignmentStart
+      val adjustment   = if (rec.getInferredInsertSize < 0) 1 else -1
+      val otherEnd     = thisEnd + rec.getInferredInsertSize + adjustment
+      val (start, end) = (min(thisEnd, otherEnd), max(thisEnd,otherEnd))
+      require(genomicPosition >= start && genomicPosition <= end, s"genomicPosition is outside of template for $rec")
```

Comment:
I agree that returning `None` where the insert size indicates that the genomic position is _outside_ the insert should be handled elsewhere.  I'd like the keep the `require` here until and have callers check (and perhaps warn) like in `FilterSomaticVcf.filterPileup`.